mr/src/main/java/org/apache/iceberg/mr/InputFormatConfig.java

```diff
@@ -58,8 +58,11 @@ private InputFormatConfig() {
   public static final String LOCATION_PROVIDER = "iceberg.mr.location.provider";
   public static final String ENCRYPTION_MANAGER = "iceberg.mr.encription.manager";
 
-  public static final String COMMIT_THREAD_POOL_SIZE = "iceberg.mr.commit.thread.pool.size";
-  public static final int COMMIT_THREAD_POOL_SIZE_DEFAULT = 10;
+  public static final String OUTPUT_TABLES = "iceberg.mr.output.tables";
+  public static final String COMMIT_TABLE_THREAD_POOL_SIZE = "iceberg.mr.commit.table.thread.pool.size";
```

Comment:
Why not use the same thread pool? And why use a thread pool at all? This could be done serially.

Reply:
I can be convinced to not do it parallel (my first patch contained no parallel execution). It really depends on the length of the Iceberg commit commit. I have seen generated BI queries with more than 10 target tables, but usually we will have only a single table as a target.

If we decide to keep the option to run the Iceberg commits in parallel, I would like to keep it as a separate pool with different sizing. So if the `iceberg.mr.commit.table.thread.pool.size` is set to `<=1`, or we have only a single target table then we will run the commit in the same thread and do not create an executor pool for the Iceberg commits.

Reply:
I would keep it simple until we have a need for parallelism. I expect the commits to be much faster than the actual execution, so I don't think that it will be a very significant delay if we do them serially. Probably better to avoid complexity until we need it later.

Reply:
> Overall, I don't think there are any blockers. I'd prefer to avoid the complexity of more thread pools and configuration constants, but it should be up to people that use Hive more than I do.

I would prefer to keep the parallel commits because I think the complexity is around the same thanks to the `Tasks` API. Other than creating the executors it is just the matter of using `Tasks` instead of loops. On the other hand since we do not have atomicity in commits we are better of committing as fast as possible to prevent concurrent changes.

I had to rebase and during the rebase I have realized that I have used the same `FileIO` to access the temp directory for every table. In the unit tests it was not an issue since all of the tables were using the same `io`, but I had to fix that.

Thanks,
Peter

Reply:
Okay, whatever you prefer. Merge when you're ready!